users/test/test_views.py

```diff
@@ -5,16 +5,105 @@
 from faker import Faker
 from rest_framework import status
 from rest_framework.reverse import reverse
-from rest_framework.test import APIRequestFactory, APITestCase
+from rest_framework.test import (
+    APIRequestFactory,
+    APITestCase,
+    force_authenticate,
+)
 
-from users.views import PasswordResetAPIView, PasswordResetEmailView
+from users.views import (
+    PasswordResetAPIView,
+    PasswordResetEmailView,
+    UserFollowView,
+)
 
 fake = Faker()
 User = get_user_model()
+follow_view = UserFollowView.as_view()
 password_reset_email_view = PasswordResetEmailView.as_view()
```

Comment:
Use the urls instead of importing the view cause that's how your app will be used

Reply:
I interchange between the two mainly because APIfactory assists in data formatting. However, even the API factory require each request to start from the URLs as would a normal request by a user.

Reply:
I can't see where you've used self.factory...apart from setUp method I only see the client, so if you are not using it remove it

Reply:
removed